src/common/cockpitjsonprint.c

```diff
@@ -28,19 +28,22 @@
 #include <fcntl.h>
 
 static bool
-char_needs_json_escape (char c)
+char_needs_json_escape (signed char c)
```

Comment:
`json_escape_string()` takes a `const char*`. I hope/guess that C does not do any funny reinterpretation between unsigned and signed char, and just leaves the bits as they are in this case?

Reply:
In case this is an unsigned character string, the individual unsigned character values will be correctly cast to signed.

Regardless of the type promotion rules, there is really only one way to imagine any signed or unsigned value, of any size, regardless of sign extension or zero extension, fitting into a signed 8 bit value: and it's the correct one.